src/datasources.js

```diff
@@ -1,3 +1,45 @@
+/**
+ * Asynchronously loads data sources from vellum.opts().core.dataSources
+ * Currently only supports fixtures
+ *
+ * Format in opts:
+ * dataSources: [
+ *     {
+ *          key: string : data source name (fixture, case, products...)
+ *          endpoint: function or string (URL)
+ *     }
+ * ]
+ *
+ * The endpoint function (or URL) should return the following format:
+ * [
+ *      {
+ *          sourceUri: string (used in the instance definition)
+ *          defaultId: string (used in instance definition)
+ *          intialQuery: string (used in nodeset)
+ *          name: string (text in the dropdown)
+ *          structure: nested dictionary of attributes
+ *          {
+ *              attribute1: {
+ *                  structure: {
+ *                      inner-attribute1: {
+ *                          no_option: boolean (does not show up in the dropdown)
+ *                      }
+ *                  }
+ *              }
+ *          }
+ *      }
+ * ]
```

Comment:
I can't think of a time you'd want to use `no_option: true` to override the default (that is, when there is a `structure` present). Do you have an example of when we would use both `structure: ...` and `no_option: ...` on the same element? If not, I feel like `no_option` is complicating the spec without adding value.